### PR TITLE
Support for Android Wear (a.k.a. Wear OS)

### LINKED ISSUE #556: Android Wear integration
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when I have to use external apps to call HA services from my Android Wear SmartWatch

**Describe the solution you'd like**
Include Android Wear support in the official HA Android app. Allow users to create Android Wear screens, up to 4 buttons on each screen, so we can call HA services from it.

**Describe alternatives you've considered, if any**
Now I'm using Tasker with AutoWear. It works, but it would be far better to only use Home Assistant Android app.

**Additional context**
Screen example: 
IMG_20200416_085746

### PR DESCRIPTION
**Pull request by Timmmeeeh**
Sorry for this huge PR, it was just an enormous undertaking to get this clean and production ready.

Fixes #556 

### Support
This PR offers support for Android Wear (a.k.a Wear OS) version 1 and 2. Running this application on version 1 requires the wearable device to always be connected to a handheld. For V2 it is only required to have your smartwatch connected to your handheld for setup of home assistant on your smartwatch. For the rest it will be completely independent operatable (iff your **smartwatch** has wifi or cellular connectivity support).

### Usage
The wear application works by defining **actions** and uses the same concept as actionable notifications but instead of `mobile_app_notification_action` a `mobile_app_wear_action` is being send with the user defined action (2nd screenshot) as data for the automation trigger.

Screenshot van Wear OS
Screenshot van Wear OS (1)
Screenshot van Wear OS (2)

### **Changes**
- Depends on #595
- **Wear Module**; Created a new module called wear. This is basically the app module for android wear.
- **Shared Module**; Some code and resources (app icons, layouts, etc.) files are used in both handheld as wear modules, by creating this shared module it will be possible to reuse any code or resources between both modules.  that were used for the phone (etc.) part, could also easily be used for the wearable part. For any file or resources suited for only that part of the application they can be included in either the app or wear module of the project.
- **Cleanup & Refactoring**; During development I noticed some Android linting errors/warnings that I've managed to fix, but also I did some general code cleanup across the application. The biggest part being some changes to the Dagger configuration, as it did not support injecting the application context in some parts that required it, like the next point below **Local Database**.
- **Local Database**; As mentioned previously actions are being fired and these actions are stored in a sqlite database. I thought about implementing it like how it's done with the widgets, but in my opinion the shared preference should not be misused for something like storing actions or widget configurations. So my suggestion is to move the widgets configuration to this database, but that is a story for another time. Room is used as database wrapper around cursors and uses a similar style as retrofit by defining queries in annotations.

### Nice to Have
- **Tests**; This PR does not include any tests written for the Android Wear application, the main reason being me not knowing the framework(/library?) that is being used. So if somebody is willing to create any tests be my guest. That being said, this PR has been tested and should be completely usable and run as expected.

### Running
If you want to run this on a wearable with Android Wear you have to make sure to follow this checklist:

1. You have to sign both of the applications (app and wear modules) with the same signing key.
2. For the wear application to be able to connect, you have to make sure that your handheld is running this version (PR) of home assistant.
3. You have setup your handheld (configure settings and login), home application app, and make sure the smartwatch is able to connect to your phone (best to check via the Wear OS app).

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -2,79 +2,139 @@
 
 ## Our Pledge
 
-In the interest of fostering an open and welcoming environment, we as
-contributors and maintainers pledge to making participation in our project and
-our community a harassment-free experience for everyone, regardless of age, body
-size, disability, ethnicity, gender identity and expression, level of experience,
-nationality, personal appearance, race, religion, or sexual identity and
-orientation.
+We as members, contributors, and leaders pledge to make participation in our
```
Comment (balloob) [inline CODE_OF_CONDUCT.md]: Please don't update the CoC

**Comment by balloob** [comment]
XKCD 1597 😛

**Comment by JBassett** [comment]
@Timmmeeeh I love that this PR exists but it might scare me a little!  I'm not sure I'm going to be able to reliably review this large of a PR.  I'm not sure if there is an easier way to break this up into smaller chunks? 
Maybe create a few PRs like:
* A general cleanup and refactoring 
  * ie, dagger
* Migrate existing widgets to database based 
  * so we can reuse this when the actual wear pr comes
* Create the generic wear module 
  * Just display something simple that has basic flow of data
  * kinda like a Proof of Concept
* Implement features in Wear module
  * this would include creating the shared module
    * can it just go in common?

I know that it might be a little more effort but I think in the end we will save time rather than have me take weeks trying to review this large of a PR.

File: app/build.gradle.kts [hunk 2]
```diff
@@ -25,7 +25,7 @@ android {
     compileSdkVersion(Config.Android.compileSdk)
 
     defaultConfig {
-        applicationId = Config.applicationId
+        applicationId = "io.homeassistant.companion.android"
```
Comment (robbiet480) [inline app/build.gradle.kts]: Don't hardcode this string

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (robbiet480) [reply CODE_OF_CONDUCT.md]: This still needs to be reverted?

File: app/build.gradle.kts [hunk 3]
```diff
@@ -52,7 +52,7 @@ android {
 
     signingConfigs {
         create("release") {
-            storeFile = file("../release_keystore.jks")
+            storeFile = file("release_keystore.keystore")
```
Comment (robbiet480) [inline app/build.gradle.kts]: Why the change?

**Comment by JBassett** [comment]
Going to close this for now.  Please feel free to reopen once ready for reconsideration.

**Comment by blair287** [comment]
Is support for wear os not coming anymore then? :(

**Comment by JBassett** [comment]
@blair287 it will come, this PR was just too massive to be merged at once.  The original creator hasn't submitted another PR since then.  If he doesn't, I'm sure someone will in time.

**Pull request closed without merging**